rust/lance-index/src/vector/ivf/shuffler.rs

```diff
@@ -348,45 +443,58 @@ impl IvfShuffler {
             let object_store = ObjectStore::local();
             let file_name = &self.unsorted_buffers[file_idx];
             let path = self.output_dir.child(file_name.as_str());
-            let reader = FileReader::try_new_self_described(&object_store, &path, None).await?;
 
-            let mut stream = stream::iter(start..end)
-                .map(|i| reader.read_batch(i as i32, ReadBatchParams::RangeFull, reader.schema()))
-                .buffered(16);
+            if self.is_legacy {
+                let reader = FileReader::try_new_self_described(&object_store, &path, None).await?;
 
-            while let Some(batch) = stream.next().await {
-                if num_processed % 100 == 0 {
-                    info!("Shuffle Progress {}/{}", num_processed, num_batches_to_sort);
-                }
-                num_processed += 1;
+                let mut stream = stream::iter(start..end)
+                    .map(|i| {
+                        reader.read_batch(i as i32, ReadBatchParams::RangeFull, reader.schema())
+                    })
+                    .buffered(16);
+
+                while let Some(batch) = stream.next().await {
+                    if num_processed % 100 == 0 {
+                        info!("Shuffle Progress {}/{}", num_processed, num_batches_to_sort);
+                    }
+                    num_processed += 1;
 
-                let batch = batch?;
+                    let batch = batch?;
 
-                // skip empty batches
-                if batch.num_rows() == 0 {
-                    continue;
+                    // skip empty batches
+                    if batch.num_rows() == 0 {
+                        continue;
+                    }
+
+                    Self::process_batch_in_shuffle(batch, &mut partitioned_batches).await?;
                 }
+            } else {
+                let scheduler = ScanScheduler::new(Arc::new(object_store));
+                let file = scheduler.open_file(&path).await?;
+                let reader = Lancev2FileReader::try_open(file, None, Default::default()).await?;
+                let mut stream = reader
+                    .read_stream(
+                        lance_io::ReadBatchParams::Range(start..end),
```

Comment:
Are `start` and `end` row numbers or batch numbers?  I thought they were batch numbers which would suggest we need a `* 1024` here (also we should maybe make that `1024` that's used in a few places now a constant `SHUFFLE_BATCH_SIZE` or something.